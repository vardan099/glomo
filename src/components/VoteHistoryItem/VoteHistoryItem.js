import React from 'react';
import PropTypes from "prop-types";
const voteTypes = {
    1: {title:"Home Team",value:"home"},
    2: {title:"Draw",value:"draw"},
    3: {title:"Away Team",value:"away"},
};

const VoteHistoryItem = (props) => {
    const {poll, voteType} =  props;
    return(
        <div className="history-item w-100">
            <div>
                <p>{`${poll.homeName} - ${poll.awayName}`}</p>
            </div>
            <div>
                <p>{poll.group} - {poll.sport}</p>
            </div>
            <div>
                <div className={`bg-${voteTypes[voteType].value} history-item_vote-type`}>
                    <span>{voteTypes[voteType].title}</span>
                </div>
            </div>
        </div>
    )
};

VoteHistoryItem.propTypes = {
    poll: PropTypes.object,
    voteType: PropTypes.number
};
export default VoteHistoryItem