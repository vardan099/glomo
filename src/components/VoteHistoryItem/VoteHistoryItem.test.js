import React from 'react';
import {shallow} from 'enzyme';
import {checkProps} from '../../TestUtils/index';
import VoteHistoryItem from './VoteHistoryItem';

describe('VoteHistoryItem Component', () => {

    describe('Checking PropTypes', () => {
        it('Should NOT throw a warning', () => {
            const expectedProps = {
                poll: jest.object,
                voteType: jest.number
            };
            const propsError = checkProps(VoteHistoryItem, expectedProps);
            expect(propsError).toBeUndefined();
        });

    });

    describe('Component Renders', () => {
        let wrapper;
        beforeEach(() => {
            const poll = {
                awayName: "Israel U18",
                homeName: "Ukraine U18",
                sport: "FOOTBALL",
                group: "Under 18",
            };
            const props ={
                poll,
                voteType: 2
            }
            wrapper = shallow(<VoteHistoryItem {...props} />);
        });

        it('renders correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });

    });


});