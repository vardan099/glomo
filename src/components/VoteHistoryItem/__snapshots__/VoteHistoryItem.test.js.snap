// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoteHistoryItem Component Component Renders renders correctly 1`] = `ShallowWrapper {}`;
